internal/controller/git/errors.go

```diff
@@ -0,0 +1,14 @@
+package git
+
+import (
+	"errors"
+)
+
+// ErrMergeConflict is returned when a merge conflict occurs.
+var ErrMergeConflict = errors.New("merge conflict")
+
+// IsMergeConflict returns true if the error is a merge conflict or wraps one
+// and false otherwise.
+func IsMergeConflict(err error) bool {
+	return errors.Is(err, ErrMergeConflict)
+}
```

Comment:
Why create an abstraction over just plain usage of `errors.Is`?

Reply:
It felt slightly more ergonomic for callers from other packages since it obviates the need to reason through whether `errors.Is()` or `errors.As()` is the appropriate approach to simply answering yes/no on whether a merge conflict occurred. In fact, with this, you don't even need to know that `ErrMergeConflict` exists. You just get a simple yes/no.

But I'm not that strongly attached to it.